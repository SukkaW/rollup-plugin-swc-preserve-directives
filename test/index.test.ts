import { tester } from './testing-utils';
import { rollup as rollup3 } from 'rollup';
import { rollup as rollup2 } from 'rollup2';

const tests = (rollupImpl: typeof rollup3) => {
  it('should preserve shebang', async () => {
    const output = await tester(rollupImpl, {
      input: 'shebang/index.ts'
    });

    expect(output).toMatchSnapshot();
  });

  it('should preserve directives', async () => {
    const output = await tester(rollupImpl, {
      input: 'use-client/index.tsx'
    });

    expect(output).toMatchSnapshot();
  });

  it('should merge duplicated directives', async () => {
    const output = await tester(rollupImpl, {
      input: 'merge-duplicated-directives/index.ts'
    });

    expect(output).toMatchSnapshot();
  });

  it('should output separate directive for multiple output chunk', async () => {
    const output = await tester(rollupImpl, {
      input: {
        client: 'separate-bundle-directives/client.tsx',
        server: 'separate-bundle-directives/server.tsx'
      }
    });

    expect(output).toMatchSnapshot();
  });

  it('issue #9', async () => {
    const output = await tester(rollupImpl, {
      input: 'prop-types/index.js'
    });

    expect(output).toMatchSnapshot();
  });
}

describe('preserve-directive (rollup 3)', () => {
  tests(rollup3);
});

describe('preserve-directive (rollup 2)', () => {
  // @ts-expect-error -- rollup 2 type is imcompatible w/ rollup 3
  tests(rollup2);
});
