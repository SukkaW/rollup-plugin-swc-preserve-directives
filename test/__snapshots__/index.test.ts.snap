// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preserve-directive (rollup 2) should merge duplicated directives 1`] = `
{
  "index": "'use client';
'use sukka';
const foo = 'foo';

const bar = 'bar';

const baz = 'baz';

export { bar, baz, foo };
",
}
`;

exports[`preserve-directive (rollup 2) should output separate directive for multiple output chunk 1`] = `
{
  "client": "'use client';
'use sukka';
import { useState } from 'react';

const foo = 'is-client';

const ClientComponent = ()=>{
    const [count] = useState(0);
    return /*#__PURE__*/ React.createElement("div", null, "count: ", count);
};

export { ClientComponent, foo };
",
  "server": "'use server';
import { readFile } from 'fs/promises';

const ServerComponent = async ()=>{
    const foo = await readFile('__virtual_data/foo.txt', 'utf8');
    return /*#__PURE__*/ React.createElement("div", null, foo);
};

export { ServerComponent };
",
}
`;

exports[`preserve-directive (rollup 2) should preserve directives 1`] = `
{
  "index": "'use client';
import { useState } from 'react';

function client() {
    return useState(null);
}

export { client as default };
",
}
`;

exports[`preserve-directive (rollup 2) should preserve shebang 1`] = `
{
  "index": "#!/usr/bin/env node
console.log('shebang');
",
}
`;

exports[`preserve-directive (rollup 3) should merge duplicated directives 1`] = `
{
  "index": "'use client';
'use sukka';
const foo = 'foo';

const bar = 'bar';

const baz = 'baz';

export { bar, baz, foo };
",
}
`;

exports[`preserve-directive (rollup 3) should output separate directive for multiple output chunk 1`] = `
{
  "client": "'use client';
'use sukka';
import { useState } from 'react';

const foo = 'is-client';

const ClientComponent = ()=>{
    const [count] = useState(0);
    return /*#__PURE__*/ React.createElement("div", null, "count: ", count);
};

export { ClientComponent, foo };
",
  "server": "'use server';
import { readFile } from 'fs/promises';

const ServerComponent = async ()=>{
    const foo = await readFile('__virtual_data/foo.txt', 'utf8');
    return /*#__PURE__*/ React.createElement("div", null, foo);
};

export { ServerComponent };
",
}
`;

exports[`preserve-directive (rollup 3) should preserve directives 1`] = `
{
  "index": "'use client';
import { useState } from 'react';

function client() {
    return useState(null);
}

export { client as default };
",
}
`;

exports[`preserve-directive (rollup 3) should preserve shebang 1`] = `
{
  "index": "#!/usr/bin/env node
console.log('shebang');
",
}
`;
